pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -210,6 +222,11 @@ public boolean index(GenericRow row) {
       // Add forward and inverted indices for new document.
       addForwardIndex(row, docId, dictIdMap);
       addInvertedIndex(docId, dictIdMap);
+
+      _lastIndexedTimestamp = System.currentTimeMillis();
```

Comment:
You also need to update it before the return in line 236 since we are aggregating the metrics there. Not sure if aggregateMetrics() silently skips aggregation, please verify.